IPython/core/debugger.py

```diff
@@ -938,10 +1031,14 @@ def do_context(self, context):
 class InterruptiblePdb(Pdb):
     """Version of debugger where KeyboardInterrupt exits the debugger altogether."""
 
-    def cmdloop(self):
+    def cmdloop(self, intro=None):
         """Wrap cmdloop() such that KeyboardInterrupt stops the debugger."""
         try:
-            return OldPdb.cmdloop(self)
+            from there import print
```

Comment:
😮 That's so cool! Wish I knew about this library before doing anything debugger related...

Reply:
Haha, it's a hack a trew together for me. I need to push it forward at some point. I'd like to have it integrate better with system logs, and have scopes. 

like:

```
def f():
    with there.print('doing stuff'):
        g()
def g():
     there.print('step 1')
     there.print('step 2')
```

And it would show a:

```
in f : doing stuff
|-  in g: step 1
|- in g: step 1
```